### PR TITLE
feat(core/webidl): support constructor linking

### LINKED ISSUE #2494: Support new constructor IDL syntax
We need to make sure we properly link and `code` constructors in a way that is backwards compatible with the old `Constructor=` extended attribute syntax. In particular `data-lt="InterfaceName.InterfaceName()"` should work, as well as `{{Foo/constructor()}}`.

### PR DESCRIPTION
**Pull request by saschanaz**
Closes https://github.com/w3c/respec/issues/394 
Closes #2494

### CODE REVIEW

File: examples/basic.html [hunk 1]
```diff
@@ -58,15 +58,19 @@ <h2>
...
+        constructor();
         attribute DOMString bar;
         void doTheFoo();
       };
       </pre>
       <p>
         The <a>Foo</a> interface is nice. Lets you do stuff.
       </p>
+      <p>
+        The <dfn>Foo.constructor()</dfn> creates a Foo instance.
```
Comment (marcoscaceres) [inline examples/basic.html]: We still want to be able to define it as `<dfn>constructor()</dfn>` like a regular operation.

Comment (saschanaz) [reply examples/basic.html]: That should work too. Will add a relevant test.

File: tests/spec/core/webidl-spec.js [hunk 2]
```diff
@@ -301,6 +301,34 @@ describe("Core - WebIDL", () => {
...
+        </pre>
+        <dfn>SuperStar.constructor()</dfn>
+        <dfn>HyperStar.constructor()</dfn>
+      </section>
+    `;
+    const ops = makeStandardOps(null, body);
+    const doc = await makeRSDoc(ops);
+    const pre = doc.querySelector("pre");
+    expect(
+      pre.querySelector("a[href=\\#dom-superstar-constructor]")
```
Comment (marcoscaceres) [inline tests/spec/core/webidl-spec.js]: what's with the "\\#dom"?

Comment (saschanaz) [reply tests/spec/core/webidl-spec.js]: That's just a common prefix we have for IDL links.

**Pull request merged**